errors/errors.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package errors
+
+import "fmt"
+
+// Error specifies an API that must be fullfiled by error type
+type Error interface {
+
+	// Error implements the error interface.
+	Error() string
+
+	// Msg returns error message
+	Msg() string
+
+	// Contains inspects if Error's message is same as error
+	// in argument. If not it continues to examin in next
+	// layers of Error until it founds it or unwrap every layers
+	Contains(error) bool
+
+	//IsEmpty check if Error is empty
+	IsEmpty() bool
+}
+
+var _ Error = (*customError)(nil)
+
+// customError struct represents a Mainflux error
+type customError struct {
+	msg string
+	err Error
+}
+
+func (err customError) Error() string {
```

Comment:
Can you rename `err` -> `ce` here and in the rest of the code please. I don't like too much this `err.err`